flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/MapJoinLeftRunner.scala

```diff
@@ -31,7 +32,19 @@ class MapJoinLeftRunner[IN1, IN2, OUT](
   override def flatMap(multiInput: IN1, out: Collector[OUT]): Unit = {
     broadcastSet match {
       case Some(singleInput) => function.join(multiInput, singleInput, out)
-      case None =>
+      case None => {
```

Comment:
A `MatchError` is thrown when no suitable pattern is found (Akka uses a lot of pattern matching, that's probably why you found that error, but it's actual a Scala thing). 

This means that there is the case of an empty `singleElement` in case of an inner join. The problem can be easily solved by not replacing `case None =>` with `case None if outerJoin =>` but just inserting `case None if outerJoin => ` in between. Pattern matching works from top to bottom and takes the first match (following matches are not executed).